src/transformers/models/bart/configuration_bart.py

```diff
@@ -168,7 +171,12 @@ def __init__(
         self.num_hidden_layers = encoder_layers
         self.gradient_checkpointing = gradient_checkpointing
         self.scale_embedding = scale_embedding  # scale factor will be sqrt(d_model) if True
-        self.force_bos_token_to_be_generated = force_bos_token_to_be_generated  # only relevant for CNN
+
+        # ensure backward compatibilty for BART CNN models
+        self.forced_bos_token_id = (
```

Comment:
Let's add a deprecation warning here and change the logic a bit:

```python
if self.forced_bos_token_id is None and kwargs.get("force_bos_token_to_be_generated", False):
    self.forced_bos_token_id = self.bos_token_id
    warnings.warn(f"...please make sure your config includes `config.forced_bos_token_id = {self.bos_token_id}` in the future. You can simply save this config and upload it again.")
```
we should manually add `forced_bos_token_id = bos_token_id` to bart-large-cnn (think this was the only model who had this logic)